submissions/Oleksandr-Heleta/OOP_exercise/index.js

```diff
@@ -0,0 +1,85 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Inhabitants {
+   constructor(species, name, gender, legs, say) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.say = say;
+   }
+}
+
+class Human extends Inhabitants {
+   constructor(name, gender, say) {
+      super('human', name, gender, 2, say);
+      this.hands = 2;
+   }
+}
+
+class Animal extends Inhabitants {
+   constructor(species, name, gender, say) {
+      super(species, name, gender, 4, say);
+   }
+}
+
+class Dog extends Animal {
+   constructor(name, gender, say) {
+      super('dog', name, gender, say);
+   }
+}
+
+class Cat extends Animal {
+   constructor(name, gender, say) {
+      super('cat', name, gender, say);
+   }
+}
+
+const dog = new Dog('Bob', "male", "bark");
+const cat = new Cat('Kitty', "female", "meow");
+
+const man = new Human('John', "male", "Hello Jenny!");
+const woman = new Human('Jenny', "female", "Hi John!");
+
+const catWoman = new Human('Jenny', "female", cat.say);
```

Comment:
Nice try here. But this way catWoman uses saying of a particular cat. Besides it will be just copied.

What is needed is that all cat-women speak like any cat and when cat's standard speak gets changed, so do cat-women.

Static class properties/methods are to the rescue.